src/main/scala/cromwell/CromwellCommandLine.scala

```diff
@@ -0,0 +1,104 @@
+package cromwell
+
+import java.nio.file.{Files, Path, Paths}
+
+import better.files._
+import cromwell.core.{ErrorOr, WorkflowSourceFiles}
+import cromwell.util.FileUtil._
+import lenthall.exception.MessageAggregation
+
+import scalaz._
+import Scalaz._
+import scala.util.{Failure, Success, Try}
+
+sealed abstract class CromwellCommandLine
+case object UsageAndExit extends CromwellCommandLine
+case object RunServer extends CromwellCommandLine
+final case class RunSingle(wdlPath: Path,
+                           sourceFiles: WorkflowSourceFiles,
+                           inputsPath: Option[Path],
+                           optionsPath: Option[Path],
+                           metadataPath: Option[Path]) extends CromwellCommandLine
+
+object CromwellCommandLine {
+  def apply(args: Seq[String]): CromwellCommandLine = {
+    args.headOption match {
+      case Some("server") if args.size == 1 => RunServer
+      case Some("run") if args.size >= 2 && args.size <= 5 => RunSingle(args.tail)
+      case _ => UsageAndExit
+    }
+  }
+}
+
+object RunSingle {
+  def apply(args: Seq[String]): RunSingle = {
+    val wdlPath = Paths.get(args.head)
+    val inputsPath = argPath(args, 1, Option(".inputs"), checkDefaultExists = false)
+    val optionsPath = argPath(args, 2, Option(".options"), checkDefaultExists = true)
```

Comment:
TBH I raised an eyebrow at it but it was working before so I brought it forward. Feature party and all that. I'll take a quick look to make sure that the code and readme at least agree.

I'd certainly having a confab regarding defaults, command line entry points,  and such at some point in the future, touching on your other point.